simulator/src/engine.js

```diff
@@ -14,11 +14,29 @@ import { showError } from './utils';
 import miniMapArea from './minimap';
 import { resetup } from './setup';
 import { verilogModeGet } from './Verilog2CV';
+import { hidetabs, visibletabs } from './listeners';
 
 /**
  * Core of the simulation and rendering algorithm.
  */
 
+/**
+ * Defining the possible states of simulator
+ * @type {Object}
+ * @category engine
+ */
+export const states = {
+    NORMAL: 'normal',
+    ERROR: 'error',
+};
+
+/**
+ * @type {string}
+ * @category engine
+ */
+const defaultState = states.NORMAL;
+export var currentState = defaultState;
+
```

Comment:
This looks good, but is there any reason for not declaring these states under scope class?

Reply:
Nope,  We can also declare this under the scope class. If you want, I can do it.

Reply:
Done 👍